browser/lib/transport/xhrrequest.ts

```diff
@@ -0,0 +1,348 @@
+import * as Utils from '../../../common/lib/util/utils';
+import EventEmitter from '../../../common/lib/util/eventemitter';
+import Platform from 'platform';
+import ErrorInfo from '../../../common/lib/types/errorinfo';
+import Http from 'platform-http';
+import Logger from '../../../common/lib/util/logger';
+import Defaults from '../../../common/lib/util/defaults';
+import * as BufferUtils from 'platform-bufferutils';
+import HttpMethods from '../../../common/constants/HttpMethods';
+import IXHRRequest from '../../../common/types/IXHRRequest';
+import { ErrnoException, RequestCallback, RequestParams } from '../../../common/types/http';
+import XHRStates from '../../../common/constants/XHRStates';
+
+// TODO replace this with the real type when Rest is in TypeScript
```

Comment:
Can this comment go now?

Reply:
[ce064d0](https://github.com/ably/ably-js/pull/762/commits/ce064d0ddb1ca40d21d618c425ec9f9eb42de62b)